sparx/bin/sxctf_refine_selection.py

```diff
@@ -0,0 +1,133 @@
+#! /usr/bin/env python
+"""
+Create substacks based on errors estimated with CTF Refinement in SPHIRE
+
+#
+# Author: Thorsten Wagner 02/18/2019 (thorsten.wagner@mpi-dortmund.mpg.de)
+# Copyright (C) 2019 Max planck institute for molecular physiology, Dortmund
+#
+# This software is issued under a joint BSD/GNU license. You may use the
+# source code in this file under either license. However, note that the
+# complete EMAN2 and SPARX software packages have some GPL dependencies,
+# so you are responsible for compliance with the licenses of these packages
+# if you opt to use BSD licensing. The warranty disclaimer below holds
+# in either instance.
+#
+# This complete copyright notice must be included in any revised version of the
+# source code. Additional authorship citations may be added, but existing
+# author citations must be preserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
+#
+#
+"""
+import argparse
+import numpy as np
+import EMAN2db
+import EMAN2
+import sxctf_refine
+import global_def
+
+ARGPARSER = argparse.ArgumentParser(
```

Comment:
Can we put the argparser part in a function calles something like "parse_args()" to avoid global variables?

Parse the args before the main function, so that main can be called from within a script.

Reply:
Done

Reply:
I put the arg parsing now into a function and call it at the beginning of main.

> Parse the args before the main function, so that main can be called from within a script.

That means I would still need the global variable ARGPARSER. I thought the goal of the change request was to get rid of it.